### PR TITLE
Clean up resampler tests

### PR DESCRIPTION
**Pull request by Luni-4**
- Compare `sys` crate results with the ones produced by the new `resampler` crate
- Clean up the tests a bit
- Fix warnings related to tests

### CODE REVIEW

File: src/resampler.rs [hunk 1]
```diff
@@ -17,43 +18,16 @@ impl fmt::Display for Error {
             Error::InvalidArg => "Invalid argument.",
             Error::PtrOverlap => "Input and output buffers overlap.",
             Error::Overflow => "Muldiv overflow.",
+            Error::UnknownError => "Unknown error.",
```
Comment (lu-zero) [inline src/resampler.rs]: This seems unrelated.

File: src/resampler.rs [hunk 2]
```diff
@@ -17,43 +18,16 @@ impl fmt::Display for Error {
...
         };
 
         write!(f, "{}", v)
     }
 }
 
-pub trait Resampler: Sized {
```
Comment (lu-zero) [inline src/resampler.rs]: This is unrelated and I'd rather not remove the trait.

File: src/resampler.rs (hunk 1 above)
Comment (Luni-4) [reply src/resampler.rs]: I added another error because I didn't want to keep the unreachable function.

File: src/echo.rs [hunk 3]
```diff
@@ -1,3 +1,5 @@
+#![allow(non_camel_case_types)]
```
Comment (lu-zero) [inline src/echo.rs]: This is not related to the patch.

File: examples/testresample.rs [hunk 4]
```diff
@@ -1,64 +0,0 @@
-use speexdsp::resampler::*;
```
Comment (lu-zero) [inline examples/testresample.rs]: Why did you delete it?

File: src/resampler.rs [hunk 5]
```diff
@@ -204,74 +204,5 @@ mod sys {
     }
 }
 
-pub mod native {
```
Comment (lu-zero) [inline src/resampler.rs]: This break the API.

**Comment by Luni-4** [comment]
Yep, let's discuss that on IRC, close this PR
